geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/SimpleFeatureSerializer.scala

```diff
@@ -466,35 +505,32 @@ object SimpleFeatureSerializer {
    * @param csFactory
    * @return
    */
-  def readGeometry(in: Input, factory: GeometryFactory, csFactory: CoordinateSequenceFactory): Geometry = {
-    if (in.readByte() == NULL_BYTE) {
-      return null
-    }
-
-    in.readInt(true) match {
-      case WKBConstants.wkbPoint => factory.createPoint(readCoordinate(in, csFactory))
+  def readGeometry(in: Input, factory: GeometryFactory, csFactory: CoordinateSequenceFactory): Geometry =
+    nullableReader( (in: Input, version: Int) => {
+      in.readInt(true) match {
+        case WKBConstants.wkbPoint => factory.createPoint(readCoordinate(in, csFactory))
 
-      case WKBConstants.wkbLineString => factory.createLineString(readCoordinateSequence(in, csFactory))
+        case WKBConstants.wkbLineString => factory.createLineString(readCoordinateSequence(in, csFactory))
 
-      case WKBConstants.wkbPolygon => readPolygon(in, factory, csFactory)
+        case WKBConstants.wkbPolygon => readPolygon(in, factory, csFactory)
 
-      case WKBConstants.wkbMultiPoint =>
-        val geoms = readGeometryCollection[Point](in, factory, csFactory)
-        factory.createMultiPoint(geoms)
+        case WKBConstants.wkbMultiPoint =>
+          val geoms = readGeometryCollection[Point](in, factory, csFactory)
+          factory.createMultiPoint(geoms)
 
-      case WKBConstants.wkbMultiLineString =>
-        val geoms = readGeometryCollection[LineString](in, factory, csFactory)
-        factory.createMultiLineString(geoms)
+        case WKBConstants.wkbMultiLineString =>
+          val geoms = readGeometryCollection[LineString](in, factory, csFactory)
+          factory.createMultiLineString(geoms)
 
-      case WKBConstants.wkbMultiPolygon =>
-        val geoms = readGeometryCollection[Polygon](in, factory, csFactory)
-        factory.createMultiPolygon(geoms)
+        case WKBConstants.wkbMultiPolygon =>
+          val geoms = readGeometryCollection[Polygon](in, factory, csFactory)
+          factory.createMultiPolygon(geoms)
 
-      case WKBConstants.wkbGeometryCollection =>
-        val geoms = readGeometryCollection[Geometry](in, factory, csFactory)
-        factory.createGeometryCollection(geoms)
-    }
-  }
+        case WKBConstants.wkbGeometryCollection =>
+          val geoms = readGeometryCollection[Geometry](in, factory, csFactory)
+          factory.createGeometryCollection(geoms)
+      }
+  })(in, 1)
```

Comment:
Oh I see . . . this application is very confusing.  Can we create some intermediate variables here to make it easier to read?